pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2877,6 +2877,23 @@ type CephFilesystemSubVolumeGroupSpec struct {
 	// list of Ceph Filesystem volumes with `ceph fs volume ls`. To learn more about Ceph Filesystem
 	// abstractions see https://docs.ceph.com/en/latest/cephfs/fs-volumes/#fs-volumes-and-subvolumes
 	FilesystemName string `json:"filesystemName"`
+	// Pinning configuration of CephFilesystemSubVolumeGroup,
+	// reference https://docs.ceph.com/en/quincy/cephfs/fs-volumes/#pinning-subvolumes-and-subvolume-groups
+	// +optional
+	Pinning CephFilesystemSubVolumeGroupSpecPinning `json:"pinning,omitempty"`
+}
+
+// CephFilesystemSubVolumeGroupSpecPinning represents the pinning configuration of SubVolumeGroup
+type CephFilesystemSubVolumeGroupSpecPinning struct {
+	// Pinning type of CephFilesystemSubVolumeGroup
+	// +optional
+	// +kubebuilder:default=distributed
+	// +kubebuilder:validation:Enum=random;export;distributed
+	PinType string `json:"pinType,omitempty"`
+	// Pinning setting of CephFilesystemSubVolumeGroup
+	// +optional
+	// +kubebuilder:default=1
```

Comment:
What are other possible pin settings? The only value I see documented in the cephfs doc on pinning is the value of `1`. If that's the only possibility, then there is no need to expose this setting. And if there are multiple possibilities, where is documentation that tells the user which value to set?

Reply:
I too feel like doc is missing its values https://docs.ceph.com/en/latest/cephfs/multimds/#manually-pinning-directory-trees-to-a-particular-rank

But Look at: https://github.com/rook/rook/pull/13283#discussion_r1407841829